modules/core/include/opencv2/core/hal/intrin_vsx.hpp

```diff
@@ -1363,8 +1364,19 @@ OPENCV_HAL_IMPL_VSX_TRANSPOSE4x4(v_uint32x4, vec_uint4)
 OPENCV_HAL_IMPL_VSX_TRANSPOSE4x4(v_int32x4, vec_int4)
 OPENCV_HAL_IMPL_VSX_TRANSPOSE4x4(v_float32x4, vec_float4)
 
-CV_CPU_OPTIMIZATION_HAL_NAMESPACE_END
+/* Permute 32B into a 16B vector using a control vector. Any control byte value greater than the number of
+   bytes in the aggregrate vector results in undefined behavior (e.x PPC will ignore those bits)  */
+inline v_uint8x16 v_permute_2(const v_uint8x16 &ctrl, const v_uint8x16 &in_lo, const v_uint8x16 &in_hi)
```

Comment:
I prefer to use uniform `v_permute` name for the intrinsic. It's possible to get quantity of permuted vectors from the list of parameters.